File: 0242-valid-anagram/0242-valid-anagram.cpp
// hashmap solution, similar to neetcode python implementation

class Solution {
public:
    bool isAnagram(string s, string t) {
        sort(s.begin(),s.end());
        sort(t.begin(),t.end());
        if(s==t)
            return true;
        else
            return false;
        
    }
};
// TC = O(NLOGN)
// SC = 0(1)
// can solve it by map TC = O(N)/O(S+T) SC= O(N)

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if(len(s)!=len(t)):
            return False
        
        cntS,cntT={},{}
        for i in range(len(s)):
            cntS[s[i]]=1+cntS.get(s[i],0)
            cntT[t[i]]=1+cntT.get(t[i],0)
        for c in cntS:
            if cntS[c]!=cntT.get(c,0):
                return False
        return True
    
    #TC = O(S+T) SC=O(N)
        
        